src/Analyzer.JsonRuleEngine/JsonRuleEngine.cs

```diff
@@ -49,6 +49,57 @@ public static JsonRuleEngine Create(string rawRuleDefinitions, BuildILineNumberR
             return new JsonRuleEngine(ParseRuleDefinitions(rawRuleDefinitions), jsonLineNumberResolverBuilder);
         }
 
+        /// <summary>
+        /// Modifies the rules to run based on values defined in the configurations file.
+        /// </summary>
+        /// <param name="configuration">The template configurations to filter and analyze.</param>
+        public void FilterRules(string configuration)
+        {
+            var filteredRules = new List<RuleDefinition>();
+
+            ConfigurationDefinition contents;
+            try
+            {
+                contents = JsonConvert.DeserializeObject<ConfigurationDefinition>(configuration);
+            }
+            catch (Exception e)
+            {
+                throw new JsonRuleEngineException($"Failed to parse configurations file.", e);
+            }
+
+            if (contents.InclusionsConfigurationDefinition != null)
+            {
+                var includeSeverities = contents.InclusionsConfigurationDefinition.Severity;
+                var includeIds = contents.InclusionsConfigurationDefinition.Ids;
+                foreach (RuleDefinition rule in RuleDefinitions)
+                {
+                    if (includeSeverities.Contains(rule.Severity) || includeIds.Contains(rule.Id))
+                    {
+                        filteredRules.Add(rule);
+                    }
+                }
+            }
+            else if (contents.ExclusionsConfigurationDefinition != null)
+            {
+                var excludeSeverities = contents.ExclusionsConfigurationDefinition.Severity;
+                var excludeIds = contents.ExclusionsConfigurationDefinition.Ids;
+
+                foreach (RuleDefinition rule in RuleDefinitions)
+                {
+                    if (!excludeSeverities.Contains(rule.Severity) || !excludeIds.Contains(rule.Id))
+                        filteredRules.Add(rule);
+                }
+            }
+            else
+            {
+                // Return original rule set - Configurations File does not have (supported) filters
+                return;
+            }
+            
+            // Update rules to select desired rules
+            RuleDefinitions = filteredRules;
```

Comment:
If you reassign this in the `if` statements, you won't need an extra `else` just to return early.